src/main/java/org/opensearch/neuralsearch/ml/MLCommonsClientAccessor.java

```diff
@@ -443,4 +458,47 @@ public void inferenceSentenceHighlighting(
     ) {
         retryableInferenceSentenceHighlighting(inferenceRequest, 0, listener);
     }
+
+    /**
+     * Execute agent with provided parameters and return DSL query string.
+     *
+     * @param agentId    the agent ID to execute
+     * @param parameters the parameters to pass to the agent
+     * @param listener   the listener to be called with the DSL query result
+     */
+    public void executeAgent(
+        @NonNull final String agentId,
+        @NonNull final Map<String, String> parameters,
+        @NonNull final ActionListener<String> listener
+    ) {
+        retryableExecuteAgent(agentId, parameters, 0, listener);
+    }
+
+    private void retryableExecuteAgent(
+        final String agentId,
+        final Map<String, String> parameters,
+        final int retryTime,
```

Comment:
what's the default times we're retrying the call before we fail?

Reply:
0